core/src/main/scala/spinal/core/internals/ComponentEmitterVhdl.scala

```diff
@@ -452,10 +453,10 @@ class ComponentEmitterVhdl(
       inc
       b ++= initialStatlementsGeneration
       dec
-      b ++= s"${tabStr}elsif ${emitClockEdge(emitReference(clock, false), clockDomain.config.clockEdge)}"
+      b ++= s"${tabStr}elsif ${emitClockEdge(emitReference(clock, false), clockDomain.config.clockEdge)} then\n"
```

Comment:
Why it is necessary to change this `then`'s position?

Reply:
To re-use the `emitClockEdge` function for the formal clocking statement and the usual process clock statement.
The formal clock statement looks like `@rising_edge(clk)`.
Previously I had a second almost identical function that also did
```scala
edgeKind match {
  case RISING  => "rising_edge"
  case FALLING => "falling_edge"
}
```
But instead of having two functions that do basically the same thing, I opted to slightly modify the existing one.
The `then` is now a the two places that also have the corresponding `if` which IMO is better to read anyway.